pkg/controller/lifecycle/actuator.go

```diff
@@ -269,6 +254,16 @@ func (a *actuator) Migrate(ctx context.Context, log logr.Logger, ex *extensionsv
 		return err
 	}
 
+	if ex.Annotations[DropDNSEntriesStateOnMigration] != "true" {
+		if err := a.ensureStateRefreshed(ctx, ex); err != nil {
+			return err
+		}
+	} else {
+		if err := a.ensureStateDropped(ctx, ex); err != nil {
+			return err
+		}
+	}
```

Comment:
Nit: negation could be avoided
```suggestion
	if ex.Annotations[DropDNSEntriesStateOnMigration] == "true" {
		if err := a.ensureStateDropped(ctx, ex); err != nil {
			return err
		}
		
	} else {
		if err := a.ensureStateRefreshed(ctx, ex); err != nil {
			return err
		}
	}
```